tools/chpl-language-server/test/type-inlays.py

```diff
@@ -0,0 +1,491 @@
+        (pos((14, 5)), "{m}Generic(int)"),
+        (pos((16, 5)), "{m}Generic({m}Concrete)"),
+        (pos((17, 5)), "{m}Generic({m}Concrete)"),
+        (pos((19, 5)), "{m}Generic({m}Generic(string))"),
+        (pos((20, 5)), "{m}Generic({m}Generic(string))"),
+    ]
+
+    variants = [("record", ""), ("class", "owned ")]
+
+    for at, management in variants:
+        with source_file(client, file.format(at=at)) as doc:
+            inlays = [
+                (pos, typename.format(m=management))
+                for pos, typename in inlay_pos
+            ]
+            await check_type_inlay_hints(
+                client, doc, rng((0, 0), endpos(file)), inlays
+            )
+            await save_file(client, doc)
+            assert len(client.diagnostics[doc.uri]) == 0
+
+
+@pytest.mark.asyncio
+async def test_type_inlays_tuple(client: LanguageClient):
+    """
+    Ensure that type inlays are shown for tuples
+    """
+
+    file = """
+            record R { }
+            var a = (10, 20);
+            var b = ((10:uint(8), ""), new R());
+            var c = b;
+            var (d, f) = (1, (2.0, new R()));
+           """
+
+    inlays = [
+        (pos((1, 5)), "(int(64), int(64))"),
+        (pos((2, 5)), "((uint(8), string), R)"),
+        (pos((3, 5)), "((uint(8), string), R)"),
+        (pos((4, 6)), "int(64)"),
+        (pos((4, 9)), "(real(64), R)"),
+    ]
+
+    with source_file(client, file) as doc:
+        await check_type_inlay_hints(
+            client, doc, rng((0, 0), endpos(file)), inlays
+        )
+        await save_file(client, doc)
+        assert len(client.diagnostics[doc.uri]) == 0
+
+
+@pytest.mark.asyncio
+@pytest.mark.xfail
+async def test_type_inlays_range(client: LanguageClient):
+    """
+    Ensure that type inlays are shown for ranges
+    """
+
+    file = """
+            var r1 = 1..10;
+            var r2 = 1..<10;
+            var r3 = 1..;
+            var r4 = ..10;
+            var r5 = ..;
+
+            var r6 = 1..10 by 10;
+            var r7 = 1..10 by -10;
+            var r8 = 1..10 by -1;
+            var r9 = 1..10 by 1;
+
+            var r10 = 1:int(32)..10:int(32);
+           """
+
+    inlays = [
+        (pos((0, 6)), "range(int(64), boundKind.both, strideKind.one)"),
+        (pos((1, 6)), "range(int(64), boundKind.both, strideKind.one)"),
+        (pos((2, 6)), "range(int(64), boundKind.low, strideKind.one)"),
+        (pos((3, 6)), "range(int(64), boundKind.high, strideKind.one)"),
+        (pos((4, 6)), "range(int(64), boundKind.neither, strideKind.one)"),
+        (pos((6, 6)), "range(int(64), boundKind.both, strideKind.positive)"),
+        (pos((7, 6)), "range(int(64), boundKind.both, strideKind.negative)"),
+        (pos((8, 6)), "range(int(64), boundKind.both, strideKind.negOne)"),
+        (pos((9, 6)), "range(int(64), boundKind.both, strideKind.one)"),
+        (pos((10, 7)), "range(int(32), boundKind.both, strideKind.one)"),
+    ]
```

Comment:
I would've expected this to work (at least in terms of picking the right `boundKind` etc), what's the problem? Is it the presentation of the type?

Reply:
The way that dyno currently prints enum values is by ID, so these print as `boundKind@2`, for example